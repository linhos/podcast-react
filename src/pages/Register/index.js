import React, {useRef, useState} from 'react';

const Register = () => {
    const registerForm = useRef(false);
    const [usernameError, setUsernameError] = useState(false);
    const [passwordError, setPasswordError] = useState(false);
    const [emailError, setEmailError] = useState(false);

    const handleRegisterFormSubmit = (event) => {
        event.preventDefault();
        console.log(`submiting...`);
        setUsernameError('El usuario es obligatorio.');
        setEmailError("El email es obligatorio.");
    }

    return (
        <>
            <h1>Register</h1>
            <form onSubmit={handleRegisterFormSubmit} ref={registerForm}>
                <label htmlFor='username'>Usuario</label>
                <input type="text" id="username" name="username" />
                {usernameError && 
                    <div data-testid="username-errors" className="error">
                        {usernameError}
                    </div>
                }

                <label htmlFor="email">Email</label>
                <input type="email" id="email" name="email" />
                {emailError && 
                    <div data-testid="email-errors" className="error">
                        {emailError}
                    </div>
                }


                <label htmlFor="password">Contrase√±a</label>
                <input type="password" id="password" name="password" required/>

                <input type="submit" value="Registro" />
            </form>
        </>
    )
}

export default Register;