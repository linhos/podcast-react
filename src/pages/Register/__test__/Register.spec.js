import React from 'react';
import { render, cleanup, fireEvent, wait } from '@testing-library/react';
import Register from '../index';
 
describe('test suite for Register component', () => {

    afterEach(cleanup);

    const mockFunctions = {
        handleRegisterFormSubmit: jest.fn(),
    }

    it('submit form when fields are empty', async () => {

        const {
            getByLabelText,
            container,
            findByTestId
        } = render(<Register />);

        const form = container.querySelector('form');
        const usernameInput = getByLabelText('Usuario');
        const emailInput = getByLabelText('Email');
        const passwordInput = getByLabelText('Contraseña');

        expect(mockFunctions.handleRegisterFormSubmit).toHaveBeenCalledTimes(0);
        
        fireEvent.blur(usernameInput);
        fireEvent.blur(emailInput);
        fireEvent.blur(passwordInput);

        fireEvent.submit(form);

        expect((await findByTestId("username-errors")).innerHTML).toBe(
            "El usuario es obligatorio."
        );

        expect((await findByTestId("email-errors")).innerHTML).toBe(
            "El email es obligatorio."
        );
    })

    it('renders', () => {
        const {asFragment} = render(<Register />);
        expect(asFragment).toMatchSnapshot();
    })

    it('form renders correctly', () => {
        const {getByLabelText, getByText} = render(<Register />);

        const usernameInput = getByLabelText('Usuario');
        expect(usernameInput).toHaveAttribute('type', 'text');

        const emailInput = getByLabelText('Email');
        expect(emailInput).toHaveAttribute('type', 'email');

        const passwordInput = getByLabelText('Contraseña');
        expect(passwordInput).toHaveAttribute('type', 'password');
        expect(passwordInput).toHaveAttribute('required');

        const submitButton = getByText('Registro');
        expect(submitButton).toHaveValue('Registro');
    })

})