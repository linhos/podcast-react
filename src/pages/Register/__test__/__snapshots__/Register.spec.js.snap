// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for Register component renders 1`] = `[Function]`;
